scikeras/wrappers.py

```diff
@@ -878,26 +893,42 @@ def partial_fit(self, X, y, sample_weight=None) -> "BaseWrapper":
         )
         return self
 
-    def predict(self, X):
+    def predict(self, X, **kwargs):
         """Returns predictions for the given test data.
 
         Parameters
         ----------
         X : Union[array-like, sparse matrix, dataframe] of shape (n_samples, n_features)
             Training samples where n_samples is the number of samples
             and n_features is the number of features.
+        **kwargs : Dict[str, Any]
+            Extra arguments to route to ``Model.predict``.
+            This paremeter will be removed in a future release of SciKeras,
```

Comment:
Why are you implementing these parameters and immediately saying this implementation will be removed in a future release?

Reply:
This is only to ease backward compatibility. We do _not_ want this `**kwargs` interface permanently. The idea would be to implement it until a `1.0.0` release or something, to help users from the old wrappers transition. Then we'd remove this interface. The alternative is to make it a required breaking change to transition to SciKeras.

Reply:
Good idea.

Style nit: I might add this text:

> **kwargs : Dict[str, Any]
Extra arguments to route to ``Model.predict``.
>
> *This functionality has been deprecated, and will be removed in SciKeras 1.0.0. These parameters can also be specified by prefixing `fit__` to a parameter at initialization; e.g, `BaseWrapper(..., fit__batch_size=32, predict__batch_size=1000)`.

I'm not sure I would deprecate support for the Keras APIs.

Reply:
> I'm not sure I would deprecate support for the Keras APIs.

I think it's generally important to have only one obvious way to do things. And in sklearn, parameters like `shuffle` and `validation_frac` are set in `__init__`, not `fit`. So that's how I think we should do it in SciKeras.

Reply:
How about the following: we change the wording from _will be deprecated_ to _is scheduled for deprecation_ and link to an issue that we can keep open for discussion. If enough good use cases come up, we can "reverse course" by simply deleting the warning in a future release. Is this a good compromise @stsievert ?

Reply:
I suggest keeping the warning, and changing the wording to "Deprecated in version v0.2.x": a behavior is deprecated if  a warning is issued by definition.

> If enough good use cases come up, we can "reverse course" by simply deleting the warning in a future release. 

Deprecations are not final. All "deprecate" means is "express disapproval of" – we can stop expressing disapproval or delete the warning if a good use case comes along.

> I think it's generally important to have only one obvious way to do things.

Touche. As a corollary, that means the error raised should suggest a resolution method.

Reply:
Sounds good, thanks for the input. I edited the warning to:
```
**kwargs has been deprecated, and will be removed in SciKeras 1.0.0.
Instead, set predict arguments at initialization (i.e., BaseWrapper({k}={v}) )
```
Where k and v are the kwarg name/key and value respectively. I think this is in line with what you have suggested.